Blackmagic/BlackMagicURLProvider.py

```diff
@@ -144,8 +144,20 @@ def compare_version(this, that):
             raise ProcessorError("Metadata for product is missing at the "
                                  "expected location in feed.")
 
-        # if this download needs registration, ensure we've set everything
-        if latest_prod["requiresRegistration"]:
+
+        # now build a request JSON to finally ask for the download URL
+        req_data = {
+            "country": "us",
+            "platform": "Mac OS X",
+            "product": {
+                "name": self.env["product_name"]
+            }
+        }
+
+        # if this download needs registration, or we want to register
+        # otherwise, ensure we've set everything
+        if latest_prod["requiresRegistration"] or \
+                self.env.get("registration_info"):
```

Comment:
I just tried this out, and it worked although it took me a couple of passes because all the recipes I tried (for things which didn't require registration) were still setting `registration_info`.. in part because some were based off of one of my recipes which sets it.

If I remove the second part of this conditional and leave only the `if latest_prod["requiresRegistration"]` as the condition, then we only enter into any of this if registration's actually required.

It ends up removing the possibility of someone sending registration info as an opt-in, but I think it simplifies the usage of recipes that would use this processor.

If a product doesn't require registration then they offer a fairly prominent `Download Only` button on the page, so I don't feel like we're going behind their back to download something that should be hidden.

Reply:
Sounds good to me.

I had already forgotten this, as I have not had anything to do with BlackMagic since late 2018... 😄 

Thanks anyway, maybe this comes handy again in near future. ;)